### PR TITLE
added trivy for vulns and sboms added cosign to sign container image and attestations upgraded docker related actions

### PR DESCRIPTION
**Pull request by garethahealy**
- signs the containers
- attachs a SBOm and Vulnerability report attestation

### CODE REVIEW

**Comment by garethahealy** [comment]
Requires: https://github.com/redhat-cop/github-workflows-operators/pull/24

**Comment by garethahealy** [comment]
TODO: i need to test it on my branch, but raising for review/comments @sabre1041 @raffaelespazzoli

**Comment by garethahealy** [comment]
Tested via: https://github.com/garethahealy/patch-operator/actions/runs/6971346429/job/18971293067

Screenshot 2023-11-23 at 15 04 33

Signed image: https://search.sigstore.dev/?logIndex=51974472

**Comment by garethahealy** [comment]
decoded attestation (vuln and sbom):
[attest.json](https://github.com/redhat-cop/github-workflows-operators/files/13451478/attest.json)

**Comment by garethahealy** [comment]
- https://github.com/redhat-cop/github-workflows-operators/issues/17
- https://github.com/redhat-cop/github-workflows-operators/issues/16

File: .github/workflows/pr-operator.yml [hunk 1]
```diff
@@ -258,6 +255,23 @@ jobs:
...
+          image-ref: ${{ env.OPERATOR_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}
+          format: "cosign-vuln"
+          output: "operator-cosignvuln.json"
+
+      - name: Run Trivy SBOM generator
+        uses: aquasecurity/trivy-action@0.14.0
+        with:
+          scan-type: image
+          image-ref: ${{ env.OPERATOR_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}
+          format: "cyclonedx"
```
Comment (sabre1041) [inline .github/workflows/pr-operator.yml:273]: Can we swap to spdx as this is the SBOM format RH is aligning to?

https://www.redhat.com/en/blog/future-red-hat-security-data

File: .github/workflows/pr-operator.yml [hunk 2]
```diff
@@ -356,6 +379,23 @@ jobs:
...
+          image-ref: ${{ env.BUNDLE_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}
+          format: "cosign-vuln"
+          output: "bundle-cosignvuln.json"
+
+      - name: Run Trivy SBOM generator
+        uses: aquasecurity/trivy-action@0.14.0
+        with:
+          scan-type: image
+          image-ref: ${{ env.BUNDLE_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}
+          format: "cyclonedx"
```
Comment (sabre1041) [inline .github/workflows/pr-operator.yml:397]: Same sbom story as above

File: .github/workflows/release-operator.yml [hunk 3]
```diff
@@ -303,6 +323,47 @@ jobs:
           push: true
           tags: "${{ env.OPERATOR_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }},${{ env.OPERATOR_IMAGE_REPOSITORY }}:${{ env.OPERATOR_VERSION }}-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}"
 
+      - name: Cosign login and initialize
+        run: |
+          echo "${{ secrets.REGISTRY_PASSWORD }}" | cosign login --username ${{ secrets.REGISTRY_USERNAME }} --password-stdin ${OPERATOR_IMAGE_REGISTRY%%/*}
+          cosign initialize
```
Comment (sabre1041) [inline .github/workflows/release-operator.yml:329]: Is initialize required if we are using the public good instance of sigstore?

File: .github/workflows/release-operator.yml [hunk 4]
```diff
@@ -457,6 +526,47 @@ jobs:
           push: true
           tags: "${{ env.BUNDLE_IMAGE_REPOSITORY }}:latest-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }},${{ env.BUNDLE_IMAGE_REPOSITORY }}:${{ env.BUNDLE_VERSION }}-${{ steps.setup-build-step.outputs.platform_os }}-${{ steps.setup-build-step.outputs.platform_arch }}"
 
+      - name: Cosign login and initialize
+        run: |
+          echo "${{ secrets.REGISTRY_PASSWORD }}" | cosign login --username ${{ secrets.REGISTRY_USERNAME }} --password-stdin ${BUNDLE_IMAGE_REGISTRY%%/*}
+          cosign initialize
```
Comment (sabre1041) [inline .github/workflows/release-operator.yml:532]: Same comment as above

File: .github/workflows/release-operator.yml (hunk 3 above)
Comment (garethahealy) [reply .github/workflows/release-operator.yml]: done

File: .github/workflows/release-operator.yml (hunk 4 above)
Comment (garethahealy) [reply .github/workflows/release-operator.yml]: done

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**